packages/edit-site/src/components/sidebar-navigation-screen-details-footer/index.js

```diff
@@ -46,7 +46,11 @@ export default function SidebarNavigationScreenDetailsFooter( {
 	return (
 		<ItemGroup className="edit-site-sidebar-navigation-screen-details-footer">
 			<SidebarNavigationItem
-				aria-label={ __( 'Revisions' ) }
+				aria-label={ sprintf(
+					/* translators: %s: human readable time difference */
+					__( 'Last modified %s' ),
```

Comment:
Whiel this solves the mismatch, I'm not sure it would be an appropriate label. This button is a navigational tool. Its label should communicate what the button does. Additional information should be moved out of the button. As such, this would need some design changes and I'd recommend to wait for some more design and accessibility feedback.

Reply:
Got it.